code/modules/mob/living/living_defense.dm

```diff
@@ -91,43 +91,77 @@
+
+	var/organ_hit_text = ""
+	if (hit_limb_zone)
+		organ_hit_text = "in \the [parse_zone_with_bodypart(hit_limb_zone)]"
+
+	switch (proj.suppressed)
+		if (SUPPRESSED_QUIET)
+			to_chat(src, span_userdanger("You're shot by \a [proj] [organ_hit_text]!"))
+		if (SUPPRESSED_NONE)
+			visible_message(span_danger("[src] is hit by \a [proj] [organ_hit_text]!"), \
+					span_userdanger("You're hit by \a [proj] [organ_hit_text]!"), null, COMBAT_MESSAGE_RANGE)
+			if(is_blind())
+				to_chat(src, span_userdanger("You feel something hit you [organ_hit_text]!"))
+
+	if(. == BULLET_ACT_HIT && proj.is_hostile_projectile())
+		apply_projectile_effects(proj, def_zone, blocked)
+
+/mob/living/proc/apply_projectile_effects(obj/projectile/proj, def_zone, armor_check)
 	apply_damage(
-		damage = hitting_projectile.damage,
-		damagetype = hitting_projectile.damage_type,
+		damage = proj.damage,
+		damagetype = proj.damage_type,
 		def_zone = def_zone,
 		blocked = min(ARMOR_MAX_BLOCK, armor_check),  //cap damage reduction at 90%
-		wound_bonus = hitting_projectile.wound_bonus,
-		bare_wound_bonus = hitting_projectile.bare_wound_bonus,
-		sharpness = hitting_projectile.sharpness,
-		attack_direction = get_dir(hitting_projectile.starting, src),
+		wound_bonus = proj.wound_bonus,
+		bare_wound_bonus = proj.bare_wound_bonus,
+		sharpness = proj.sharpness,
+		attack_direction = get_dir(proj.starting, src),
 	)
+
 	apply_effects(
-		stun = hitting_projectile.stun,
-		knockdown = hitting_projectile.knockdown,
-		unconscious = hitting_projectile.unconscious,
-		slur = (mob_biotypes & MOB_ROBOTIC) ? 0 SECONDS : hitting_projectile.slur, // Don't want your cyborgs to slur from being ebow'd
-		stutter = (mob_biotypes & MOB_ROBOTIC) ? 0 SECONDS : hitting_projectile.stutter, // Don't want your cyborgs to stutter from being tazed
-		eyeblur = hitting_projectile.eyeblur,
-		drowsy = hitting_projectile.drowsy,
+		stun = proj.stun,
+		knockdown = proj.knockdown,
+		unconscious = proj.unconscious,
+		slur = (mob_biotypes & MOB_ROBOTIC) ? 0 SECONDS : proj.slur, // Don't want your cyborgs to slur from being ebow'd
+		stutter = (mob_biotypes & MOB_ROBOTIC) ? 0 SECONDS : proj.stutter, // Don't want your cyborgs to stutter from being tazed
+		eyeblur = proj.eyeblur,
+		drowsy = proj.drowsy,
 		blocked = armor_check,
-		stamina = hitting_projectile.stamina,
-		jitter = (mob_biotypes & MOB_ROBOTIC) ? 0 SECONDS : hitting_projectile.jitter, // Cyborgs can jitter but not from being shot
-		paralyze = hitting_projectile.paralyze,
-		immobilize = hitting_projectile.immobilize,
+		stamina = proj.stamina,
+		jitter = (mob_biotypes & MOB_ROBOTIC) ? 0 SECONDS : proj.jitter, // Cyborgs can jitter but not from being shot
+		paralyze = proj.paralyze,
+		immobilize = proj.immobilize,
 	)
-	if(hitting_projectile.dismemberment)
-		check_projectile_dismemberment(hitting_projectile, def_zone)
+
+	if(proj.dismemberment)
+		check_projectile_dismemberment(proj, def_zone)
+
 	return BULLET_ACT_HIT
```

Comment:
Why does this return `BULLET_ACT_HIT`? Doesn't seem like the ret val is used anywhere